### PR TITLE
feat(excel): support Workbooks and Tables

### PR DESCRIPTION
**Pull request by dominique-pfister**

### CODE REVIEW

File: src/OneDrive.d.ts [hunk 1]
```diff
@@ -110,6 +240,8 @@ export declare class OneDrive extends EventEmitter {
 
   downloadDriveItem(driveItem: DriveItem): Promise<GraphResult>;
 
+  getWorkbook(fileItem: DriveItem): Workbook;
```
Comment (tripodsan) [inline src/OneDrive.d.ts:243]:
```suggestion
  /**
    * Creates a new `Workbook` instance for the given drive item.
    */
  getWorkbook(fileItem: DriveItem): Workbook;
```

File: src/Table.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
```
Comment (tripodsan) [inline src/Table.js:2]:
```suggestion
 * Copyright 2020 Adobe. All rights reserved.
```

File: src/OneDrive.d.ts [hunk 3]
```diff
@@ -43,6 +43,136 @@ export declare interface SubscriptionOptions {
...
+   * @returns zero-based index of new row
+   */
+  addRow(values: string[]): Promise<number>;
+
+  /**
+   * Replace a row in the table
+   * @param index zero-based index of row
+   * @param values values to replace rows with
+   */
+  replaceRow(index: string, values: string): void;
```
[the remark this answers is not in the document]
Comment (dominique-pfister) [reply src/OneDrive.d.ts:96]: It should 😄

File: src/OneDrive.d.ts [hunk 4]
```diff
@@ -43,6 +43,136 @@ export declare interface SubscriptionOptions {
...
+   * Return the worksheet names contained in a work book.
+   * @returns array of sheet names when resolved
+   */
+  getWorksheetNames(): Promise<string[]>;
+
+  /**
+   * Return a work sheet given its name
+   * @param name work sheet name
+   */
+  getWorksheet(name: string): Worksheet;
```
[the remark this answers is not in the document]
Comment (dominique-pfister) [reply src/OneDrive.d.ts:139]: 👍

**Comment by dominique-pfister** [comment]
Increased test coverage for newly added classes.

File: src/NamedItemContainer.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
```
Comment (tripodsan) [inline src/NamedItemContainer.js:2]:
```suggestion
 * Copyright 2020 Adobe. All rights reserved.
```

File: src/OneDrive.d.ts [hunk 6]
```diff
@@ -43,6 +43,148 @@ export declare interface SubscriptionOptions {
...
+ */
+export declare interface Workbook {
+  /**
+   * Return the worksheet names contained in a work book.
+   * @returns array of sheet names when resolved
+   */
+  getWorksheetNames(): Promise<string[]>;
+
+  /**
+   * Return a work sheet given its name
```
Comment (tripodsan) [inline src/OneDrive.d.ts:142]:
```suggestion
   * Return a new worksheet instance given its name
```

File: src/OneDrive.d.ts [hunk 7]
```diff
@@ -43,6 +43,148 @@ export declare interface SubscriptionOptions {
...
+  worksheet(name: string): Worksheet;
+
+  /**
+   * Return the table names contained in a work book.
+   * @returns array of table names when resolved
+   */
+  getTableNames(): Promise<string[]>;
+
+  /**
+   * Return a table given its name
```
Comment (tripodsan) [inline src/OneDrive.d.ts:154]:
```suggestion
   * Return a new `Table` instance given its name
```

File: src/Workbook.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
```
Comment (tripodsan) [inline src/Workbook.js:2]:
```suggestion
 * Copyright 2020 Adobe. All rights reserved.
```

**Pull request merged**